recipes/zstd/all/test_package/CMakeLists.txt

```diff
@@ -1,17 +1,13 @@
-cmake_minimum_required(VERSION 3.1)
-project(test_package C)
+cmake_minimum_required(VERSION 3.15)
+project(test_package LANGUAGES C)
 
 file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/logo.png
      DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
-
 find_package(zstd REQUIRED)
 
 add_executable(${PROJECT_NAME} test_package.c)
-if (TARGET zstd::libzstd_shared)
-  target_link_libraries(${PROJECT_NAME} zstd::libzstd_shared)
-else()
-  target_link_libraries(${PROJECT_NAME} zstd::libzstd_static)
-endif()
+
+# Note: assuming that only one of these targets exists
+target_link_libraries(${PROJECT_NAME} $<TARGET_NAME_IF_EXISTS:zstd::libzstd_shared>)
```

Comment:
😱 https://cmake.org/cmake/help/v3.12/manual/cmake-generator-expressions.7.html

Such a nice feature

Reply:
this syntax is horrible, especially when you need to write a complex condition
more it looks way too different and foreign over an existing regular `if` conditions
I really don't like it, my eyes are bleeding to read it, compare to:
```
if (TARGET zstd::libzstd_shared)
 target_link_libraries(${PROJECT_NAME} zstd::libzstd_shared)
endif()
```